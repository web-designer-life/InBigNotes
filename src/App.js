import React from 'react';
import './App.css';
import Container from './Container';

class App extends React.Component {
  render() {
      return (
          <div>
              <Container />
          </div>
      );
  }
}

export default App;
