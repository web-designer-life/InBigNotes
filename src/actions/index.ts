export const FETCH_NOTES_PENDING = 'FETCH_NOTES_PENDING';
export const FETCH_NOTES_SUCCESS = 'FETCH_NOTES_SUCCESS';
export const FETCH_NOTES_FAIL = 'FETCH_NOTES_FAIL';

export const ADD_NOTES_PENDING = 'ADD_NOTES_PENDING';
export const ADD_NOTES_SUCCESS = 'ADD_NOTES_SUCCESS';
export const ADD_NOTES_FAIL = 'ADD_NOTES_FAIL';

export const DELETE_NOTES_PENDING = 'DELETE_NOTES_PENDING';
export const DELETE_NOTES_SUCCESS = 'DELETE_NOTES_SUCCESS';
export const DELETE_NOTES_FAIL = 'DELETE_NOTES_FAIL';

export default { 
    FETCH_NOTES_PENDING,
    FETCH_NOTES_SUCCESS,
    FETCH_NOTES_FAIL,
    ADD_NOTES_PENDING, 
    ADD_NOTES_SUCCESS, 
    ADD_NOTES_FAIL, 
    DELETE_NOTES_PENDING, 
    DELETE_NOTES_SUCCESS, 
    DELETE_NOTES_FAIL,
};
