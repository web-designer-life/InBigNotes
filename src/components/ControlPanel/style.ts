import styled from 'styled-components';

const Wrapper = styled.div`
    display: flex;
    justify-content: space-between;
    padding: 0 50px;
`;

const SelectAllWrapper = styled.div`
    display: flex;
    justify-content: space-between;
    align-items: center;
`;

const Title = styled.p`
    font-style: normal;
    font-weight: bold;
    font-size: 36px;
    line-height: 46px;
    margin: 0 30px 0 0;
    color: #5A5A5A;
`;

export { 
    Wrapper, 
    SelectAllWrapper, 
    Title,
};
