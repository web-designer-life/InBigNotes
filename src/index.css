body {
    font-family: 'Lora', 'Arial', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
    height: 100%;
    margin: 0;
    padding: 0;
}

code {
    font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New', monospace;
}

#root {
    display: flex;
    flex-direction: column;
    max-width: 1280px;
    height: 100%;
    margin: auto;
}
