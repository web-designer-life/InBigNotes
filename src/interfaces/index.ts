export interface INote {
    id: string;
    title: string;
    text: string;
    created_at: number;
    updated_at: number;
};
