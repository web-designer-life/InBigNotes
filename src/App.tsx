import * as React from 'react';
import { GlobalStyle } from './Components/Style/GlobalStyle';
import Container from './Container';
import './App.css';

class App extends React.Component {
    render() {
        return (
            <>
                <GlobalStyle />
                <Container />
            </>
        );
    }
};
  
export default App;
