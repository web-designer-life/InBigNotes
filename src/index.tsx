import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import history, { store } from './store/store';
import App from './containers/App';

ReactDOM.render(
    <React.StrictMode>
        <Provider store={store}>
            <App history={history} />
        </Provider>
    </React.StrictMode>,
    document.getElementById('root')
);
