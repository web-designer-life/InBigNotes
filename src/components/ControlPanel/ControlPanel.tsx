import * as React from 'react';
import { Note } from '../../interfaces';
import Checkbox from '../Checkbox/Checkbox';
import Filter from '../Filter/Filter';
import { 
    Wrapper, 
    SelectAllWrapper, 
    Title,
} from './style';

interface Props {
    notes?: Note[],
    selectedNotes?: string[],
    selectAllNotes(notes: Note[]): Function,
    unselectAllNotes(): Function,
}

export default class ControlPanel extends React.Component<Props> {
    constructor(props: Props) {
        super(props);

        this.handleSelectOrUnselectNotes = this.handleSelectOrUnselectNotes.bind(this);
    };

    handleSelectOrUnselectNotes() {
        const { 
            notes, 
            selectedNotes,
            selectAllNotes, 
            unselectAllNotes,
        } = this.props;

        selectedNotes?.length === notes?.length ?
        unselectAllNotes() :
        selectAllNotes(notes || [])
    };

    render() {
        const { 
            notes, 
            selectedNotes,
        } = this.props;

        return (
            <Wrapper>
                <Filter />
                <SelectAllWrapper>
                    <Title>Select all {selectedNotes?.length ? `(${selectedNotes?.length})` : ''}</Title>
                    <Checkbox
                        checked={selectedNotes?.length === notes?.length}
                        onChange={this.handleSelectOrUnselectNotes}
                    />
                </SelectAllWrapper>
            </Wrapper>
        )
    }
};
