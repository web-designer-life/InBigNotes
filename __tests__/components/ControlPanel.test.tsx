import React from 'react';
import { shallow } from 'enzyme';
import ControlPanel from '../../src/components/ControlPanel';
import { INote } from '../../src/interfaces';

const controlPanelProps = {
    notes: [
        {
            id: 'b83cbac2-d74e-458d-85c0-75bdf9c29ba3',
            title: 'Army',
            text: 'An army, ground force, or land force is a fighting force that fights primarily on land.',
            created_at: 1632734670806,
            updated_at: 1632735522467,
        },
    ],
    selectedNotes: [
        'b83cbac2-d74e-458d-85c0-75bdf9c29ba3',
    ],
    filterType: 'title',
    fetchNotes: (filterType: string) => Function,
    selectAllNotes: (notes: INote[]) => Function,
    unselectAllNotes: () => Function,
    filterAction: (filterType: string) => Function,
};

const controlPanelPropsWithoutSelectedNotes = {
    notes: [
        {
            id: 'b83cbac2-d74e-458d-85c0-75bdf9c29ba3',
            title: 'Army',
            text: 'An army, ground force, or land force is a fighting force that fights primarily on land.',
            created_at: 1632734670806,
            updated_at: 1632735522467,
        },
    ],
    selectedNotes: [],
    filterType: 'title',
    fetchNotes: (filterType: string) => Function,
    selectAllNotes: (notes: INote[]) => Function,
    unselectAllNotes: () => Function,
    filterAction: (filterType: string) => Function,
};

describe('ControlPanel component', () => {
    it('should render ControlPanel component', () => {
        const component = shallow(<ControlPanel {...controlPanelProps} />);
        expect(component).toMatchSnapshot();
    });

    it('should render ControlPanel component without selectedNotes', () => {
        const component = shallow(<ControlPanel {...controlPanelPropsWithoutSelectedNotes} />);
        expect(component).toMatchSnapshot();
    });
});
