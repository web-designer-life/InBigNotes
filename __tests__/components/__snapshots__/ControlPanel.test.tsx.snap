// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlPanel component should render ControlPanel component 1`] = `
<styled.div>
  <Filter
    fetchNotes={[Function]}
    filterAction={[Function]}
    filterType="title"
  />
  <styled.div>
    <styled.p>
      Select all 
      (1)
    </styled.p>
    <Checkbox
      checked={true}
      onChange={[Function]}
    />
  </styled.div>
</styled.div>
`;

exports[`ControlPanel component should render ControlPanel component without selectedNotes 1`] = `
<styled.div>
  <Filter
    fetchNotes={[Function]}
    filterAction={[Function]}
    filterType="title"
  />
  <styled.div>
    <styled.p>
      Select all 
    </styled.p>
    <Checkbox
      checked={false}
      onChange={[Function]}
    />
  </styled.div>
</styled.div>
`;
